GameServerCore/Domain/IInventoryManager.cs

```diff
@@ -1,11 +1,18 @@
-﻿namespace GameServerCore.Domain
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace GameServerCore.Domain
 {
     public interface IInventoryManager
     {
         IItem GetItem(byte slot);
         void RemoveItem(byte slot);
+        void RemoveItem(IItem item);
+        IItem AddItem(IItemType item);
         byte GetItemSlot(IItem item);
         IItem SetExtraItem(byte slot, IItemType item);
         void SwapItems(byte slot1, byte slot2);
+        List<IItem> GetAvailableItems(IItemRecipe recipe);
```

Comment:
Because it only needs the recipe to get the items from. I mean sure I can use IItemType here and call the Recipe property if you think thats better.

Reply:
But why inventoryManager need recipe to get the AvailableItems?, what does this function should do?

Reply:
Aah now I see what you mean, yes it makes no sense to me now either. It currently fetches all items of that recipe and checks what of those the champion already has. I am gonna change it so that you should pass a list of items to look through instead.

Reply:
Aah now I see what you mean, yes it makes no sense to me now either. It currently fetches all items of that recipe and checks what of those the champion already has. I am gonna change it so that you should pass a list of items to look through instead.